### DISCUSSION TITLE
PDF - Custom module > Related Custom module with with line items

### DISCUSSION THREAD

**Discussion by s2b-git** [YetiForceCompany/YetiForceCRM#15860] [How to]
Somehow struggling to create report as follow:
Custom Module > Related Advanced Custom module with line items.
PDF Templater contains variable "Table with advanced block for related modules" but it not works
My variable format: $(custom : ProductsTableRelatedModule|CustomModule|custommoduleid|seq,name,qty,discount,currency,discountmode,taxmode,price,gross,net,tax,total)$

Am I missing something?
One remark. This shall apply to Services module, but I guess it doesn't make difference?

**Comment by mariuszkrzaczkowski** [comment]
Give an example to better understand what you mean

**Comment by s2b-git** [comment]
Please look here:
https://gitdeveloper.yetiforce.com/index.php?module=Accounts&view=List&mid=51&parent=47
PDF template:
test-rel-inv[ test-rel-inv

**Reply by mariuszkrzaczkowski** [reply]
I have a PDF like this:
[PDF.pdf](https://github.com/YetiForceCompany/YetiForceCRM/files/8416307/PDF.pdf)

There is no such "ProductsTableRelatedModule" function in the system, the list of functions is here:
https://github.com/YetiForceCompany/YetiForceCRM/blob/b65343dacf96cb3fad3dc3b5c2ffdf6a220c08b7/app/TextParser.php#L97

I still don't understand what kind of PDF you want

**Comment by s2b-git** [comment]
![image](https://user-images.githubusercontent.com/84110093/161712532-9edf0e36-eb63-495e-8e85-56b0a160c47d.png)


best

**Reply by mariuszkrzaczkowski** [reply]
I entered the code incorrectly
is
`ProductsTableRelatedModule|Invoice|invoiceid|seq,name,qty,discount,currency,discountmode,taxmode,price,gross,net,tax,total)$`
and it should be
`$(ProductsTableRelatedModule|Invoice|invoiceid|seq,name,qty,discount,currency,discountmode,taxmode,price,gross,net,tax,total)$`

Try to do this example for the "Payments in" module:
`$(custom : ProductsTableRelatedModule|FInvoiceProforma|finvoiceproformaid|seq,name,qty,discount,currency,discountmode,taxmode,price,gross,net,tax,total)$`

**Reply by s2b-git** [reply]
tried. Not working
best!

**Reply by mariuszkrzaczkowski** [reply]
Be specific about how you test, what you do, what data !
If not, I will not waste my time on it !!!

**Reply by s2b-git** [reply]
I retested in local instance, and everything in correctly set. Basically variable does not returning list of records. None showing up.
Will work on it and if I will find solution will share it here

best

**Reply by mariuszkrzaczkowski** [reply]
Check it in the Payments in module,

`
$(custom : ProductsTableRelatedModule|FInvoiceProforma|finvoiceproformaid|seq,name,qty,discount,currency,discountmode,taxmode,price,gross,net,tax,total)$
`

finvoiceproformaid - field of relation to the related module FInvoiceProforma
FInvoiceProforma - the module related to which we want to show the data
